app/src/main/java/org/redcross/openmapkit/MapActivity.java

```diff
@@ -135,27 +159,40 @@ protected void positionMap() {
         double lat = (double) pref.getFloat(PREVIOUS_LAT, -999);
         double lng = (double) pref.getFloat(PREVIOUS_LNG, -999);
         float z = pref.getFloat(PREVIOUS_ZOOM, -999);
-        
+
         // no shared pref
         if (lat == -999 || lng == -999 || z == -999) {
             mapView.setUserLocationEnabled(true);
             mapView.goToUserLocation(true);
-        } 
+            if (z == -999) {
+                mapView.setUserLocationRequiredZoom(50);
+            } else {
+                mapView.setUserLocationRequiredZoom(z);
+            }
+        }
         // there is a shared pref
         else {
             LatLng c = new LatLng(lat, lng);
             mapView.setCenter(c);
             mapView.setZoom(z);
+            mapView.setUserLocationEnabled(true);
         }
     }
-    
+
+    protected void addGeojsonOverlay() {
+        mapView.loadFromGeoJSONURL("http://api.mspray.onalabs.org/buffers.json?target_area=408_257");
```

Comment:
Or Nick feel free to remove that and merge it in.  Thanks!

On Sun, Jul 19, 2015 at 8:47 PM, Matt Berg <mberg@ona.io> wrote:

> Apologies this is something that shouldn't be here.  Was a test of mapbox
> sdk geojson rendering.  Will ask Isaac to revert and do another pull
> request.
>
> On Sun, Jul 19, 2015 at 8:39 PM, Nicholas Hallahan <
> notifications@github.com> wrote:
>
>> In app/src/main/java/org/redcross/openmapkit/MapActivity.java
>> <https://github.com/AmericanRedCross/OpenMapKit/pull/70#discussion_r34960157>
>> :
>>
>> >          }
>> >      }
>> > -
>> > +
>> > +    protected void addGeojsonOverlay() {
>> > +        mapView.loadFromGeoJSONURL("http://api.mspray.onalabs.org/buffers.json?target_area=408_257");
>>
>> What is the hard coded JSON here? Should we put this in a settings file
>> maybe?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/AmericanRedCross/OpenMapKit/pull/70/files#r34960157>.
>>
>
>